### DISCUSSION TITLE
Problem with Image::newFromFile

### DISCUSSION THREAD

**Discussion by KirbyDE** [libvips/libvips#2473] [Q&A]
I am not sure if this is a bug or if I am doing smth. wrong, but can't find a way to re-read an image:

```php
require('./vendor/autoload.php');

use Jcupitt\Vips\Image;

$file = '/tmp/original.jpg';
$tempFile = tempnam('/tmp', 'vips');
$image = Image::newFromFile($file, ['autorotate' => false]);
echo('Dimensions original: ' . $image->width . ' x ' . $image->height . "\n");

$image = $image->rotate(90);
echo('Dimensions after rotation: ' . $image->width . ' x ' . $image->height . "\n");

$image->jpegsave($tempFile, ['strip' => true]);

$image = Image::newFromFile($tempFile, ['autorotate' => true]);
echo('Dimensions tempFile with autorot: ' . $image->width . ' x ' . $image->height . "\n");

$image = Image::newFromFile($tempFile, ['autorotate' => false]);
echo('Dimensions tempFile without autorot: ' . $image->width . ' x ' . $image->height . "\n");

unlink($file);
rename($tempFile, $file);

$image = Image::newFromFile($file, ['autorotate' => true]);
echo('Dimensions file with autorot: ' . $image->width . ' x ' . $image->height . "\n");

$image = Image::newFromFile($file, ['autorotate' => false]);
echo('Dimensions file without autorot: ' . $image->width . ' x ' . $image->height . "\n");

$image = Image::newFromBuffer(file_get_contents($file));
echo('Dimensions buffer: ' . $image->width . ' x ' . $image->height . "\n");
```

I  had expected that the output for tempFile and and file after save would be identical, but they are not:

> Dimensions original: 1480 x 720
Dimensions after rotation: 720 x 1480
Dimensions tempFile with autorot: 720 x 1480
Dimensions tempFile without autorot: 720 x 1480
Dimensions file with autorot: 720 x 1480
Dimensions file without autorot: 1480 x 720
Dimensions buffer: 720 x 1480

Does anyone have an idea what is going on here?

**Comment by jcupitt** [comment]
Hi @KirbyDE,

libvips caches things like `newFromFile`. If you give the same arguments as before, it'll give you the old result again (and save a decompression). It does this everywhere, and it can sometimes save a lot of time.

I think your line:

```php
rename($tempFile, $file);
```

Is confusing the caching system.

I would not reuse filenames. If you must use a filename again, you can either disable the libvips operation cache with `cacheSetMax(0)` (this will probably hurt speed), or you can prevent an operation being cached by loading from a buffer rather than a file.
